pkg/clean/clean.go

```diff
@@ -48,15 +44,14 @@ var Backoff = wait.Backoff{
 // GetNSXClientFailed  			indicate that could not retrieve nsx client to perform cleanup operation
 // InitCleanupServiceFailed 	indicate that error happened when trying to initialize cleanup service
 // CleanupResourceFailed    	indicate that the cleanup operation failed at some services, the detailed will in the service logs
-func Clean(ctx context.Context, cf *config.NSXOperatorConfig, logg *logr.Logger, debug bool, logLevel int) error {
-	// Clean may be called many times, only init once
-	once.Do(func() {
-		if logg != nil {
-			logf.SetLogger(*logg)
-		} else {
-			logf.SetLogger(logger.ZapLogger(debug, logLevel))
-		}
-	})
+func Clean(ctx context.Context, cf *config.NSXOperatorConfig, log *logr.Logger, debug bool, logLevel int) error {
+	// Clean needs to support many instances which each have its own logger
+	if log == nil {
+		logg := logger.ZapLogger(debug, logLevel)
+		log = &logg
+	}
+	logf.SetLogger(*log)
```

Comment:
I feel this is strange for a library, unrelated to logging, to set a global logger. I think this is better set in `cmd_clean` and omitted here.

Reply:
remove it.